hack/pre-commit.sh

```diff
@@ -24,6 +24,11 @@ function run_check() {
 
     if [ -x "$(command -v "$exe")" ]; then
         echo "=====  $exe  ====="
+        if ! "$scriptdir"/version_test_ge_xyz.sh "$1" "$2"; then
+            echo >&2 error: version precedes minimum
```

Comment:
Do we really need to print the error here?

The check version scripts have more context and can log a more helpful
message like:

    error: version 0.7.0 < minimum 0.7.2

We will use the check version script in many other places. It will be
easier to use if the script handles the error output.